app/Events/VoucherHistoryEmailRequested.php

```diff
@@ -18,17 +18,23 @@ class VoucherHistoryEmailRequested
 
     public $user;
     public $trader;
+    public $vouchers;
+    public $date;
+    public $max_date;
     public $file;
 
     /**
      * Create a new event instance.
      *
      * @return void
      */
-    public function __construct(User $user, Trader $trader, $file)
+    public function __construct(User $user, Trader $trader, $vouchers, $file, $date, $max_date = null)
```

Comment:
You are pushing a lot of params in here. Is there a more modular way to do this?